Source/SwiftLintFramework/Rules/Performance/EmptyArrayRule.swift

```diff
@@ -0,0 +1,33 @@
+import SourceKittenFramework
+
+public struct EmptyArrayRule: ConfigurationProviderRule, OptInRule, AutomaticTestableRule {
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "empty_array",
+        name: "Empty Array",
+        description: "Prefer checking `isEmpty` over comparing `array` to an empty array literal.",
+        kind: .performance,
+        nonTriggeringExamples: [
+            "myArray = []",
+            "myArray.isEmpty",
+            "!myArray.isEmpy"
+        ],
+        triggeringExamples: [
+            "myArray↓ == []",
+            "myArray↓ != []"
+        ]
+    )
+
+    public func validate(file: File) -> [StyleViolation] {
+        let pattern = "\\b\\s*(==|!=)\\s*\\[\\]"
```

Comment:
What about the case `myArray == [ ]`? Is that covered by this pattern?

Reply:
Good catch, updating now!